### DISCUSSION TITLE
ETF holdings analytics

### DISCUSSION THREAD

**Discussion by Kampvognen** [ghostfolio/ghostfolio#3149] [Ideas]
Hi all,

Sharing an idea of more extensive analytics on the portfolio composition when holding ETFs.

**Rationale:** 
When holding several ETFs, one can be unsure of exposure overlap of underlying assets which can come from different ETFs holding the same assets.
**Why is this important:** 
Over-exposure can lead to concentration rather than diversification. Therefore, analyzing the overlaps between different ETFs can help make more informed portfolio decisions
**Where can we see this work in practice:** 
MorningStars free portfolio tool XRay with stock overlap feature.

Out of all the other platforms I've tried I've only seen Morningstar nail this **while** also having a more complete ETF list. Ghostfolio has the latter which is why I think it'd be amazing with more analytics on the underlying asset levels to ensure I'm not overexposed to certain companies. Sector level is for me too high level and only tells a story about high level diversification and not the investment theme in more detail.

**Comment by dtslvr** [comment]
This feature has been released as experimental with the latest version (see [Ghostfolio 2.86.0](https://github.com/ghostfolio/ghostfolio/releases/tag/2.86.0)).

**Reply by JoryHogeveen** [reply]
Hi @dtslvr 
I was about to post a duplicate discussion but found this page.

Could you elaborate how to view/enable the ETF overlap feature?
I can see the allocations for what ETF's I own and in what countries these are invested but I cannot see the holdings they include or any overlap of these ETFs.

**Reply by dtslvr** [reply]
Hi @JoryHogeveen

Experimental features can be enabled via _My Ghostfolio_. Then you will see the holdings at the bottom of the allocations page.

**Reply by JoryHogeveen** [reply]
@dtslvr 
Awesome, thank you!

This is already very insightful.
Few suggestions I have which I think would be relatively easy to add:
- Add insights from what ETF the holding is from and for what percentage
- In case this is available, add some insights of the holding (industry and country for example)

Though I'm not very familiar with Angular, this is JS so I might able to contribute if pointed in the right direction to get started.

**Reply by dtslvr** [reply]
The first idea seems feasible, but there's no data available for the second at the moment.

What you could is the following:

- Extend the type of `topHoldings` in the [allocations page](https://github.com/ghostfolio/ghostfolio/blob/main/apps/client/src/app/pages/portfolio/allocations/allocations-page.component.ts#L89) by parent holdings:
```
interface HoldingWithParents extends Holding {
    parents: Holding[];
}

public topHoldings: HoldingWithParents[];
```
- Adjust the [calculation of topHoldings](https://github.com/ghostfolio/ghostfolio/blob/main/apps/client/src/app/pages/portfolio/allocations/allocations-page.component.ts#L479) (via `topHoldingsMap`)
- Pass `topHoldings` (with parent holdings) to [GfTopHoldingsComponent](https://github.com/ghostfolio/ghostfolio/blob/main/libs/ui/src/lib/top-holdings/top-holdings.component.ts#L44)
- Set up [expandable rows](https://material.angular.io/components/table/examples#table-expandable-rows) inside of [GfTopHoldingsComponent](https://github.com/ghostfolio/ghostfolio/blob/main/libs/ui/src/lib/top-holdings/top-holdings.component.html) to list the parent holdings including the same columns (_Name_, _Value_, _Allocation_).

Could you do anything with these hints?

**Reply by JoryHogeveen** [reply]
Thank you for that, I will give it a shot!

**Reply by JoryHogeveen** [reply]
@dtslvr I gave it a shot! https://github.com/ghostfolio/ghostfolio/pull/4044

See PR for notes and ideas for optional improvements.

Btw, compliments on the easy setup for development!
